internal/controller/node_controller.go

```
@@ -47,11 +47,12 @@ func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	nodeObj := &corev1.Node{}
 	err := r.Get(ctx, req.NamespacedName, nodeObj)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			r.Log.WithValues("name", req.Name).Info("node not found")
-			return ctrl.Result{}, nil
-		}
+	if apierrors.IsNotFound(err) {
+		r.Log.WithValues("name", req.Name).Info("node not found")
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		r.Log.WithValues("name", req.Name).Error(err, "error fetching object")
+		return ctrl.Result{}, err
```

Get node failed and returned an error to retry, instead of continuing to execute downwards